### PR TITLE
chore: update @kubernetes/client-node to v1.0.0

### PR DESCRIPTION
**Pull request by feloy**
### What does this PR do?

Update @kubernetes/client-node to v1.0.0 and removes previous patches.

Do not mark the library as external anymore as the format has been chaged to ESM: https://github.com/kubernetes-client/javascript/pull/2062

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -197,6 +197,7 @@
     "hpagent": "^1.2.0",
     "minimist": "^1.2.8",
     "moment": "^2.30.1",
+    "node-fetch": "^3.3.2",
```
Comment (benoitf) [inline package.json]: issue: fetch is now part of nodejs it should not be added with a 3rd party library

Comment (feloy) [reply package.json]: I added it as I got an error when commit'ing the changes. I'll check again

Comment (feloy) [reply package.json]: I finally removed the use of the `AbortError` type, to use the same check as in the library:

https://github.com/kubernetes-client/javascript/blob/release-1.x/src/health.ts#L52

```
if (err instanceof Error && err.name === 'AbortError') {
```

File: packages/main/src/plugin/kubernetes/context-health-checker.ts [hunk 2]
```diff
@@ -111,4 +111,8 @@ export class ContextHealthChecker implements Disposable {
   public getState(): ContextHealthState {
     return this.#currentState;
   }
+
+  private isAbortError(err: unknown): boolean {
```
Comment (axel7083) [inline packages/main/src/plugin/kubernetes/context-health-checker.ts:115]:
```suggestion
  private isAbortError(err: unknown): err is AbortError {
```

Comment (feloy) [reply packages/main/src/plugin/kubernetes/context-health-checker.ts:115]: But if we do this, we need again the AbortError type. I wanted to get rid of it

Comment (axel7083) [reply packages/main/src/plugin/kubernetes/context-health-checker.ts:115]: :/ okey

**Review by axel7083 — APPROVED** [review]
LGTM

File: packages/main/vite.config.js [hunk 3]
```diff
@@ -51,7 +51,6 @@ const config = {
       external: [
         'electron',
         'chokidar',
-        '@kubernetes/client-node',
```
Comment (benoitf) [inline packages/main/vite.config.js]: highly recommended: I would remove this part from this pull request as it's not directly related to 1.0.0-RC7 to 1.0.0

**Review by benoitf — APPROVED** [review]

File: packages/main/vite.config.js (hunk 3 above)
Comment (feloy) [reply packages/main/vite.config.js]: I think it is related, as the library has been changed to ESM between the 2 versions.

Comment (benoitf) [reply packages/main/vite.config.js]: but doing it there and now can have side consequences in production mode.

so it's always preferable to not mix two things at a single time

Comment (feloy) [reply packages/main/vite.config.js]: OK, I've created https://github.com/podman-desktop/podman-desktop/pull/10439

Comment (jeffmaury) [reply packages/main/vite.config.js]: So this change should be removed ?

Comment (feloy) [reply packages/main/vite.config.js]: yes, it will be removed after rebase

**Pull request merged**